extensions/data-transfer/portability-data-transfer-imgur/src/test/java/org/datatransferproject/transfer/imgur/photos/ImgurPhotoExporterTest.java

```diff
@@ -0,0 +1,272 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ImgurPhotoExporterTest {
+
+  private MockWebServer server;
+  private OkHttpClient client = new OkHttpClient.Builder().build();
+  private ObjectMapper mapper = new ObjectMapper();;
+  private TokensAndUrlAuthData token =
+      new TokensAndUrlAuthData("accessToken", "refreshToken", "tokenUrl");
+  private JobStore jobStore = mock(JobStore.class);
+  private ImgurPhotosExporter exporter;
+  private Monitor monitor = mock(Monitor.class);
+
+  // contains albums
+  private String albumsResponse;
+  // contains photos from the first album
+  private String album1ImagesResponse;
+  // contains both album and non-album photos
+  private String allImagesResponse;
+  // contains photos for paged requests
+  private String page1Response;
+  private String page2Response;
+
+  {
+    try {
+      albumsResponse =
+          Resources.toString(Resources.getResource("albums.json"), Charsets.UTF_8).trim();
+      album1ImagesResponse =
+          Resources.toString(Resources.getResource("album_1_images.json"), Charsets.UTF_8).trim();
+      allImagesResponse =
+          Resources.toString(Resources.getResource("all_images.json"), Charsets.UTF_8).trim();
+      page1Response =
+          Resources.toString(Resources.getResource("page1.json"), Charsets.UTF_8).trim();
+      page2Response =
+          Resources.toString(Resources.getResource("page2.json"), Charsets.UTF_8).trim();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    server = new MockWebServer();
+    server.start();
+    exporter =
+        new ImgurPhotosExporter(monitor, client, mapper, jobStore, server.url("").toString());
+  }
+
+  @Test
+  public void testAlbumsExport() throws Exception {
+    server.enqueue(new MockResponse().setBody(albumsResponse));
+
+    // export albums
+    ExportResult<PhotosContainerResource> result =
+        exporter.export(UUID.randomUUID(), token, Optional.empty());
+    PhotosContainerResource resource = result.getExportedData();
+
+    assertEquals(2, resource.getAlbums().size());
+    assertEquals(0, resource.getPhotos().size());
+
+    PhotoAlbum album1 =
+        resource
+            .getAlbums()
+            .stream()
+            .filter(album -> "albumId1".equals(album.getId()))
+            .findFirst()
+            .get();
+    assertEquals("Album 1", album1.getName());
+
+    PhotoAlbum album2 =
+        resource
+            .getAlbums()
+            .stream()
+            .filter(album -> "albumId2".equals(album.getId()))
+            .findFirst()
+            .get();
+    assertEquals("Album 2", album2.getName());
```

Comment:
This is nice. Do you think we can condense the code from 105 and 108-124 if we use the Truth library?

I was thinking something along the lines of #3 here (https://google.github.io/truth/)  using assertThat() containsExactly() and inOrder() along with your filter for album id?

Reply:
I updated some tests to use Truth library. All the code from lines 115-135 could be changed to 
```
    assertThat(resource.getAlbums())
        .containsExactly(
            new PhotoAlbum("albumId1", "Album 1", null),
            new PhotoAlbum("albumId2", "Album 2", "Description for Album 2")).inOrder();
```
But PhotoAlbum's [equals method](https://github.com/google/data-transfer-project/blob/1e31ec8a70b04ac6ceee017e6a51b805d57aa16a/portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoAlbum.java#L58) checks only if `id`s are equal. So need to check name and description separately.
PhotoModels are compared by all fields so I condensed their tests.